src/Interpreters/MutationsInterpreter.cpp

```diff
@@ -655,7 +660,10 @@ void MutationsInterpreter::prepare(bool dry_run)
                 if (materialized_it != column_to_affected_materialized.end())
                     for (const auto & mat_column : materialized_it->second)
                         affected_materialized.emplace(mat_column);
+            }
 
+            for (const auto & [column_name, update_expr] : command.column_to_update_expression)
```

Comment:
IIUC, this is necessary in order to collect all affected_materialized before starting size_checks